packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../application_package.dart';
+import '../base/file_system.dart';
+import '../globals.dart';
+import '../ios/plist_utils.dart' as plist;
+
+ /// Tests whether a [FileSystemEntity] is an macOS bundle directory
+bool _isBundleDirectory(FileSystemEntity entity) =>
+    entity is Directory && entity.path.endsWith('.app');
+
+abstract class MacOSApp extends ApplicationPackage {
+  MacOSApp({@required String projectBundleId}) : super(id: projectBundleId);
+
+   /// Creates a new [MacOSApp] from an existing app bundle.
```

Comment:
A little more detail here could be good. How is it going to do that?

Reply:
Done